services/navigation/builtin/builtin.go

```diff
@@ -400,19 +400,18 @@ func (svc *builtIn) startWaypointExperimental(extra map[string]interface{}) erro
 		defer svc.activeBackgroundWorkers.Done()
 
 		navOnce := func(ctx context.Context, wp navigation.Waypoint) error {
-			currentLoc, err := svc.Location(svc.cancelCtx, extra)
-			if err != nil {
-				return err
+			if extra != nil && extra["motion_profile"] == nil {
+				extra["motion_profile"] = "position_only"
+			} else {
+				extra = map[string]interface{}{"motion_profile": "position_only"}
```

Comment:
If extra is non-nil, has all sorts of important stuff in it, including a `"motion_profile"` set to anything other than nil, then this will overwrite the *entire* `extra` object.

IMO this would be better written as:

```
if extra == nil {
	extra = map[string]interface{}{"motion_profile": "position_only"}
} else if extra["motion_profile"] == nil {
	extra["motion_profile"] = "position_only"
}
```
That will prevent the whole data structure from being overwritten, and if the user went out of their way to select a motion profile, it is preserved rather than being silently overwritten.